command/agent/agent.go

```diff
@@ -440,6 +440,14 @@ func (a *Agent) consulConfig() *consul.Config {
 		base.AutopilotConfig.DisableUpgradeMigration = *a.config.Autopilot.DisableUpgradeMigration
 	}
 
+	// set the src address for outgoing rpc connections
+	// to RPCAdvertise with port 0 so that outgoing
+	// connections use a random port.
+	// todo(fs): can base.RPCAdvertise be nil ?
+	base.RPCSrcAddr = new(net.TCPAddr)
```

Comment:
This is another place where the configuration gets modified: https://github.com/hashicorp/consul/blob/master/consul/server.go#L228-L230